### ISSUE TITLE
Latest 2.8.0 SNAPSHOT fails in dev mode on JDK7 due to being built with JDK8 without an appropriate bootclasspath

### ISSUE THREAD

**Issue by cgdecker** [gwtproject/gwt#9206]
All GWT tests for [Guava](https://github.com/google/guava) running on JDK7 started failing (unable to complete) with the latest 2.8.0 SNAPSHOT that was released on October 20th 2015 (`gwt-dev-2.8.0-20151020.054017-77.jar`). The failure (which you can see [here](https://travis-ci.org/google/guava/jobs/86655833#L4901)) is as follows:

```
Running com.google.gwt.junit.tools.GWTTestSuite@4e460434
[ERROR] Unable to load module entry point class null (see associated exception for details)
java.lang.NoClassDefFoundError: java/lang/reflect/Executable
        at com.google.gwt.dev.shell.JavaDispatchImpl.getMethod(JavaDispatchImpl.java:122)
        <...>
```

I did some investigation into what could be causing this and found [this post](http://www.draconianoverlord.com/2014/04/01/jdk-compatibility.html) which contains a likely explanation (and is even about the exact issue that's occurring here):

The GWT snapshot was compiled with JDK8 using `-source -target` less than 8 but WITHOUT providing an appropriate `bootclasspath` (i.e. a JDK7 bootclasspath for `-source 7 -target 7`). When this happens, the JDK8 compiler can sneak references to JDK8 types (`java.lang.reflect.Executable` in this case) into the bytecode, because it finds them on the classpath and doesn't know they aren't included in the version they're targeting. Thus the code will run on the lower JDK due to the `-target` version but fail when it hits the JDK8 type in the bytecode.

Sure enough, the most recent GWT 2.8.0 snapshot was built with JDK8, while the previous snapshot was built with JDK7. The solution, as indicated in the linked article, is either to go back to building with JDK7 or to set the `bootclasspath` appropriately when building with JDK8.

**Comment by tbroyer** [comment]
/cc @gkdn @dankurka @manolo
We should move the main build job back to Java 7.

**Comment by gkdn** [comment]
It is still blurry but GWT 2.8 will very likely not support dev mode with java7 (there were also other problems with that IIRC). Also my anticipation is Guava will stop supporting dev mode in their next release as well - internally we are no longer testing guava with devmode. I'll need to talk to @cpovirk to clarify that and how to proceed with this issue.

**Comment by gkdn** [comment]
BTW, to mitigate the issue for now, Roberto is changing back gwt build to java 7 so we will keep building SDK releases with java 7. He is also changing gwt.java8 build to send emails to the contributor list.

**Comment by gkdn** [comment]
Can you verify if the problem is resolved?

**Comment by cgdecker** [comment]
Thanks... I would have talked to @cpovirk about this first, but he's out yesterday and today.

**Comment by cgdecker** [comment]
Oh, just saw your question about verifying that it's resolved... I'll check that in a bit.

**Comment by cgdecker** [comment]
Yes, it seems to be resolved... thanks!

**Comment by cpovirk** [comment]
Desupporting Dev Mode SGTM.

I intend to clean up our code for GWT 2.8.0 within the next few weeks. I'll remove Guava's Dev Mode tests at that point. Let me know if you'd like them to be removed sooner.

**Comment by gkdn** [comment]
There is no urgency from our side. We will probably still have some limited level of support for devmode in the next release. I'll let you know if the plans change.

**Comment by cpovirk** [comment]
Some combination of Goktug, me, and perhaps other have wiped out Dev Mode support and testing from Guava 20.0, to be released shortly after GWT 2.8.0. (It will both support and require GWT 2.8.0.)
